dlp/src/main/java/dlp/snippets/InspectGcsFileWithSampling.java

```diff
@@ -0,0 +1,174 @@
+              .setFileSet(FileSet.newBuilder().setUrl(gcsUri))
+              .setBytesLimitPerFile(200)
+              .addFileTypes(FileType.TEXT_FILE)
+              .setFilesLimitPercent(90)
+              .setSampleMethod(SampleMethod.RANDOM_START)
+              .build();
+
+      StorageConfig storageConfig =
+          StorageConfig.newBuilder().setCloudStorageOptions(cloudStorageOptions).build();
+
+      // Specify the type of info the inspection will look for.
+      // See https://cloud.google.com/dlp/docs/infotypes-reference for complete list of info types
+      InfoType infoType = InfoType.newBuilder().setName("PERSON_NAME").build();
+
+      // Specify how the content should be inspected.
+      InspectConfig inspectConfig =
+          InspectConfig.newBuilder()
+              .addInfoTypes(infoType)
+              .setExcludeInfoTypes(true)
+              .setIncludeQuote(true)
+              .setMinLikelihood(Likelihood.POSSIBLE)
+              .build();
+
+      // Specify the action that is triggered when the job completes.
+      String pubSubTopic = String.format("projects/%s/topics/%s", projectId, topicId);
+      Action.PublishToPubSub publishToPubSub =
+          Action.PublishToPubSub.newBuilder().setTopic(pubSubTopic).build();
+      Action action = Action.newBuilder().setPubSub(publishToPubSub).build();
+
+      // Configure the long running job we want the service to perform.
+      InspectJobConfig inspectJobConfig =
+          InspectJobConfig.newBuilder()
+              .setStorageConfig(storageConfig)
+              .setInspectConfig(inspectConfig)
+              .addActions(action)
+              .build();
+
+      // Create the request for the job configured above.
+      CreateDlpJobRequest createDlpJobRequest =
+          CreateDlpJobRequest.newBuilder()
+              .setParent(LocationName.of(projectId, "global").toString())
+              .setInspectJob(inspectJobConfig)
+              .build();
+
+      // Use the client to send the request.
+      final DlpJob dlpJob = dlp.createDlpJob(createDlpJobRequest);
+      System.out.println("Job created: " + dlpJob.getName());
+
+      // Set up a Pub/Sub subscriber to listen on the job completion status
+      final SettableApiFuture<Boolean> done = SettableApiFuture.create();
+
+      ProjectSubscriptionName subscriptionName =
+          ProjectSubscriptionName.of(projectId, subscriptionId);
+
+      MessageReceiver messageHandler =
+          (PubsubMessage pubsubMessage, AckReplyConsumer ackReplyConsumer) -> {
+            handleMessage(dlpJob, done, pubsubMessage, ackReplyConsumer);
+          };
+      Subscriber subscriber = Subscriber.newBuilder(subscriptionName, messageHandler).build();
+      subscriber.startAsync();
+
+      // Wait for the original job to complete
+      try {
+        done.get(15, TimeUnit.MINUTES);
+      } catch (TimeoutException e) {
+        System.out.println("Job was not completed after 15 minutes.");
```

Comment:
Is there something a user should do here?

Reply:
For now I'm keeping this consistent with the other existing samples. Obviously "println and return" is probably not the right answer for most prod systems, but it does keep the sample simple. Hopefully it's easy for users to tune to their own needs. 

As a broader effort, it would be great to have a consistent approach to recommend across samples, across APIs, and (ideally) across languages as well.